kubernetes/eks-setup.sh

```diff
@@ -0,0 +1,150 @@
+     --profile dnschg
+
+# For MQTTS_FQDN
+
+DNS_RECORD_NAME=$(aws acm describe-certificate --certificate-arn $CERTIFICATE_ARN --profile or --query "Certificate.DomainValidationOptions[1].ResourceRecord.Name")
+DNS_RECORD_VALUE=$(aws acm describe-certificate --certificate-arn $CERTIFICATE_ARN --profile or --query "Certificate.DomainValidationOptions[1].ResourceRecord.Value")
+
+aws route53 change-resource-record-sets \
+    --hosted-zone-id /hostedzone/Z08751721JH0NB6LLCB4V \
+    --change-batch \
+     '{"Changes": [ { "Action": "UPSERT", "ResourceRecordSet": { "Name": '$DNS_RECORD_NAME', "Type": "CNAME", "TTL": 300, "ResourceRecords" : [ { "Value": '$DNS_RECORD_VALUE' } ] } } ]}' \
+     --profile dnschg
+
+# AWS LB Controller only creates an Application LB if there's an ingress
+# Following logic is assuming there is the only (application) LB in the account
+while ! aws elbv2 describe-load-balancers  --profile or --query "LoadBalancers[?Type=='application']" 2>/dev/null | grep '"Code": "active"'; do
+  echo "Waiting for application load balancer to be created..."
+  sleep 10
+done
+
+# We're re-directing the FQDN to the application LB (web interface)
+DNS_NAME=$(aws elbv2 describe-load-balancers --profile or --query "LoadBalancers[?Type=='application'].DNSName | [0]")
+HOSTED_ZONE_ID=$(aws elbv2 describe-load-balancers --profile or --query "LoadBalancers[?Type=='application'].CanonicalHostedZoneId | [0]")
+
+aws route53 change-resource-record-sets \
+    --hosted-zone-id /hostedzone/Z08751721JH0NB6LLCB4V \
+    --change-batch \
+     '{"Changes": [ { "Action": "UPSERT", "ResourceRecordSet": { "Name": "'$FQDN'", "Type": "A", "AliasTarget":{ "HostedZoneId": '$HOSTED_ZONE_ID',"DNSName": '$DNS_NAME',"EvaluateTargetHealth": false} } } ]}' \
+     --profile dnschg
+
+# We're re-directing the MQTT_FQDN to its network load balancer
+DNS_NAME=$(kubectl get svc manager-mqtt -o jsonpath="{.status.loadBalancer.ingress[0].hostname}")
+while ! aws elbv2 describe-load-balancers --profile or --query "LoadBalancers[?DNSName=='$DNS_NAME']" 2>/dev/null | grep '"Code": "active"'; do
+  echo "Waiting for network load balancer to be created for $DNS_NAME..."
+  sleep 10
+done
+HOSTED_ZONE_ID=$(aws elbv2 describe-load-balancers --profile or --query "LoadBalancers[?DNSName=='$DNS_NAME'].CanonicalHostedZoneId | [0]")
+
+aws route53 change-resource-record-sets \
+    --hosted-zone-id /hostedzone/Z08751721JH0NB6LLCB4V \
+    --change-batch \
+     '{"Changes": [ { "Action": "UPSERT", "ResourceRecordSet": { "Name": "'$MQTT_FQDN'", "Type": "A", "AliasTarget":{ "HostedZoneId": '$HOSTED_ZONE_ID',"DNSName": '\"$DNS_NAME\"',"EvaluateTargetHealth": false} } } ]}' \
+     --profile dnschg
+
+# We're re-directing the MQTTS_FQDN to its network load balancer
+DNS_NAME=$(kubectl get svc manager-mqtts -o jsonpath="{.status.loadBalancer.ingress[0].hostname}")
```

Comment:
While testing the `DNS_NAME` was an empty string causing the subsequent while loop to never exit:

`Waiting for network load balancer to be created for ...`

To fix that you might want to change the code so it waits until the command is not empty or add a quick fix like `sleep 3s`